File: URITemplateTests/RFC6570/RFC6570Level2Tests.m
//
//  RFC6570Level2Tests.m
//  URITemplate
//
//  Created by Emlyn Murphy on 9/24/15.
//  Copyright © 2015 Emlyn Murphy. All rights reserved.
//

#import <XCTest/XCTest.h>
#import <URITemplate/URITemplate.h>

@interface RFC6570Level2Tests : XCTestCase

@property (nonatomic, strong) NSDictionary *values;

@end

@implementation RFC6570Level2Tests

- (void)setUp {
    [super setUp];

    self.values = @{ @"var": @"value",
                     @"hello": @"Hello World!",
                     @"path": @"/foo/bar" };
}

- (void)tearDown {
    self.values = nil;

    [super tearDown];
}

- (void)testLevel2Example1 {
    URITemplate *template = [URITemplate templateWithString:@"{+var}"];
    NSString *expected = @"value";
    NSString *result = [template expand:self.values];

    XCTAssert([expected isEqualToString:result]);
}

- (void)testLevel2Example2 {
    URITemplate *template = [URITemplate templateWithString:@"{+hello}"];
    NSString *expected = @"Hello%20World!";
    NSString *result = [template expand:self.values];

    XCTAssert([expected isEqualToString:result]);
}

- (void)testLevel2Example3 {
    URITemplate *template = [URITemplate templateWithString:@"{+path}/here"];
    NSString *expected = @"/foo/bar/here";
    NSString *result = [template expand:self.values];

    XCTAssert([expected isEqualToString:result]);
}

- (void)testLevel2Example4 {
    URITemplate *template = [URITemplate templateWithString:@"here?ref={+path}"];
    NSString *expected = @"here?ref=/foo/bar";
    NSString *result = [template expand:self.values];

    XCTAssert([expected isEqualToString:result]);
}

@end

File: URITemplateTests/RFC6570Level2Tests.swift
//
//  RFC6570Level2Tests.swift
//  URITemplate
//
//  Created by Emlyn Murphy on 8/22/15.
//  Copyright (c) 2015 Emlyn Murphy. All rights reserved.
//
//
//  Level 2 examples from RFC 6570
//
//  URL: https://tools.ietf.org/html/rfc6570
//

import UIKit
import XCTest
import URITemplate

class RFC6570Level2ExampleTests: XCTestCase {
    var values: URITemplateValues!

    override func setUp() {
        super.setUp()

        values = [
            "var": "value",
            "hello": "Hello World!",
            "path": "/foo/bar"
        ]
    }

    func testLevel2Examples() {
        let reservedStringURITemplate1 = URITemplate("{+var}")
        let reservedStringExpected1 = "value"
        let reservedStringResult1 = reservedStringURITemplate1.expand(values)

        XCTAssert(reservedStringExpected1 == reservedStringResult1,
            "expected \"\(reservedStringExpected1)\"; got \"\(reservedStringResult1)\"")

        let reservedStringURITemplate2 = URITemplate("{+hello}")
        let reservedStringExpected2 = "Hello%20World!"
        let reservedStringResult2 = reservedStringURITemplate2.expand(values)

        XCTAssert(reservedStringExpected2 == reservedStringResult2,
            "expected \"\(reservedStringExpected2)\"; got \"\(reservedStringResult2)\"")

        let reservedStringURITemplate3 = URITemplate("{+path}/here")
        let reservedStringExpected3 = "/foo/bar/here"
        let reservedStringResult3 = reservedStringURITemplate3.expand(values)

        XCTAssert(reservedStringExpected3 == reservedStringResult3,
            "expected \"\(reservedStringExpected3)\"; got \"\(reservedStringResult3)\"")

        let reservedStringURITemplate4 = URITemplate("here?ref={+path}")
        let reservedStringExpected4 = "here?ref=/foo/bar"
        let reservedStringResult4 = reservedStringURITemplate4.expand(values)

        XCTAssert(reservedStringExpected4 == reservedStringResult4,
            "expected \"\(reservedStringExpected4)\"; got \"\(reservedStringResult4)\"")
    }
}
